x-pack/plugins/ingest_manager/server/routes/data_streams/handlers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { RequestHandler } from 'src/core/server';
+import { DataStream } from '../../types';
+import { GetDataStreamsResponse } from '../../../common';
+
+const DATA_STREAM_INDEX_PATTERN = 'logs-*-*,metrics-*-*';
+
+export const getListHandler: RequestHandler = async (context, request, response) => {
+  const callCluster = context.core.elasticsearch.dataClient.callAsCurrentUser;
+
+  try {
+    // Get stats (size on disk) of all potentially matching indices
+    const { indices: indexStats } = await callCluster('indices.stats', {
+      index: DATA_STREAM_INDEX_PATTERN,
+      metric: ['store'],
+    });
+
+    // Get all matching indices and info about each
+    // This returns the top 100,000 indices (as buckets) by last activity
+    const {
+      aggregations: {
+        index: { buckets: indexResults },
+      },
+    } = await callCluster('search', {
+      index: DATA_STREAM_INDEX_PATTERN,
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            must: [
+              {
+                exists: {
+                  field: 'fields.stream.namespace',
+                },
+              },
+              {
+                exists: {
+                  field: 'fields.stream.dataset',
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          index: {
+            terms: {
+              field: '_index',
+              size: 100000,
+              order: {
+                last_activity: 'desc',
+              },
+            },
+            aggs: {
+              dataset: {
+                terms: {
+                  field: 'fields.stream.dataset',
+                  size: 1,
+                },
+              },
+              namespace: {
+                terms: {
+                  field: 'fields.stream.namespace',
+                  size: 1,
+                },
+              },
+              type: {
+                terms: {
+                  field: 'fields.stream.type',
+                  size: 1,
+                },
+              },
+              package: {
+                terms: {
+                  field: 'event.module',
```

Comment:
Why is this needed? Sorting? Perhaps prefix query could help here? `event.module` is likely to disappear in the future.

Reply:
This is to detect the package to display in the Integrations column of the table. I see the same value in `service.type` too? What should I use?

Reply:
None of both :-( This brings up the question: Is it the data that links to the package or does the ingest manager know which package belongs to the data? I would say it is the second one. Lets assume someone configure the agent to ship system metrics. He will not have added any information around packages as the agent does not know about packages. But the data will still be processed correctly, stored and works with the dashboards. Basically for each stream the ingest manager must check if it belongs to an installed package to show the link and then also show the dashboards ...

If your current solution works, lets keep it and then iterate on it.

Reply:
Ok, it sounds like the get the package info the "right" way will involve more work that overlaps with the work needed for linking to dashboard, so will defer this for now.

Does that mean this PR gets a 👍? 🙂

Reply:
The query gets a 👍 Did not really check the rest of the code or did it run locally.